utilities/map.metta

```diff
@@ -39,6 +39,24 @@
 (= (Map.remove $key (ConsMap ($curKey $curVal) $tail)) 
     (if (== $key $curKey) $tail (ConsMap ($curKey $curVal) (Map.remove $key $tail))))
 
+;; Get the all the values from the map
+(: Map.values (-> (Map ($v $v)) Expression))
```

Comment:
As a convention, we prefer returning `(List $v)` instead of `Expression`. And also the type of map should not be `(Map ($v $v))`. That will force the map to have the same type for keys and values which isn't always the case.